tests/test_utils.py

```diff
@@ -167,3 +167,37 @@ def test_is_overdue_false_if_within_grace_period(self, subject, a, stub_config):
         timeline = subject(a.participant(), reference_time, stub_config)
 
         assert not timeline.is_overdue
+
+
+class TestBaseURL(object):
+    @pytest.fixture
+    def subject(self):
+        from dallinger.utils import get_base_url
+
+        return get_base_url
+
+    @pytest.fixture
+    def config(self):
+        instance = config.get_config()
+        instance.ready = True
+        yield instance
+        config.config = None
+
+    def test_local_base_url(self, subject, config):
+        config.set("host", u"localhost")
+        config.set("base_port", 5000)
+        config.set("num_dynos_web", 1)
+        assert subject() == u"http://localhost:5000"
+
+    def test_remote_base_url_always_ssl(self, subject, config):
+        config.set("host", u"http://dlgr-bogus.herokuapp.com")
+        config.set("base_port", 80)
+        config.set("num_dynos_web", 1)
+        assert subject() == u"https://dlgr-bogus.herokuapp.com"
+
+    def test_os_HOST_environ_used_as_host(self, subject, config):
+        with mock.patch("os.environ", {"HOST": u"dlgr-bogus-2.herokuapp.com"}):
```

Comment:
I came across this recently:
https://docs.pytest.org/en/latest/monkeypatch.html#monkeypatching-environment-variables

Maybe we should use it here?

Reply:
I'd be inclined against adding a dependency. The advantage of that pattern is that it lets you set items individually, but this is okay with just overwriting the list and restoring it.

Reply:
My understanding is that the `monkeypatch` fixture is part of pytest, so no new dependency. I agree what we're doing here is fine; the argument for `monkeypatch` would be to model a new way of doing this going forward.